Signal/src/UIView+OWS.h

```diff
@@ -0,0 +1,51 @@
+//
+//  Copyright (c) 2017 Open Whisper Systems. All rights reserved.
+//
+
+#import <UIKit/UIKit.h>
+
+// A convenience method for doing responsive layout. Scales between two
+// reference values (for iPhone 5 and iPhone 7 Plus) to the current device
+// based on screen width, linearly interpolating.
+CGFloat ScaleFromIPhone5To7Plus(CGFloat iPhone5Value, CGFloat iPhone7PlusValue);
+
+// A convenience method for doing responsive layout. Scales a reference
+// value (for iPhone 5) to the current device based on screen width,
+// linearly interpolating through the origin.
+CGFloat ScaleFromIPhone5(CGFloat iPhone5Value);
+
+// A set of helper methods for doing layout with PureLayout.
+@interface UIView (OWS)
+
+// Pins the width of this view to the width of its superview, with uniform margins.
+- (void)autoPinWidthToSuperviewWithMargin:(CGFloat)margin;
+- (void)autoPinWidthToSuperview;
```

Comment:
Ideally, these convenience methods don't just DRY up repetitive layout code, their names are higher-level and more semantic than raw AL operations.